### PR TITLE
Remove non-core adapters

### PR DESCRIPTION
**Pull request by reinecke**
If there is an associated issue, link it in the form:

This PR covers the major work of #1386.

Use one line for each Issue. This allows auto-closing the related issue when the fix is merged.

**Summarize your change.**

This PR applies core repo cleanup after creating separate repos for adapters in the [OpenTimelineIO GitHub org](https://github.com/OpenTimelineIO).

- Removes contrib adapters and tree
- Removes all but the "native" OTIO adapters from `py-opentimelineio/opentimelineio/adapters`
- Updates to plugin system and documentation to support adapter removal

This PR is the result of contributions from the broad community including:

- @apetrynet 
- @markreidvfx
- @rosborne132 
- @timlehr

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -11,9 +11,9 @@ env:
 
 on:
   push:
-    branches: [ main ]
+    branches: [ main, extract_adapters ]
   pull_request:
-    branches: [ main ]
+    branches: [ main, extract_adapters ]
```
Comment (apetrynet) [inline .github/workflows/python-package.yml:16]: Same thought as in docs.yaml. I think we should only keep "main"

File: README.md [hunk 2]
```diff
@@ -73,6 +73,9 @@ For more information: https://github.com/AcademySoftwareFoundation/OpenTimelineI
 
 For more information about this, including supported formats, see: https://opentimelineio.readthedocs.io/en/latest/tutorials/adapters.html
 
+The **AAF Adapter** has been **relocated** to a separate repository and is located here:
+https://github.com/OpenTimelineIO/otio-aaf-adapter
+
 Other Plugins
 -------------
```
Comment (apetrynet) [inline README.md:81]:
```suggestion
All adapters except the native `.otio`, `.otioz` and `.otiod` have been relocated to separate repositories under the OpenTimelineIO organization located here: https://github.com/OpenTimelineIO
```

File: docs/tutorials/adapters.md [hunk 3]
```diff
@@ -2,82 +2,3 @@
 
 OpenTimelineIO supports, or plans to support, conversion adapters for many
 existing file formats.
```
Comment (apetrynet) [inline docs/tutorials/adapters.md:4]: There's not much left of this file.. do we still need it or should we at least flesh it out a bit?

Perhaps add links to each adapter repo?

**Comment by reinecke** [comment]
Build failing due to scikit-build/cmake-python-distributions/pull/490

File: setup.py [hunk 4]
```diff
@@ -366,7 +363,6 @@ def run(self):
     python_requires='>=3.7, !=3.9.0',  # noqa: E501
 
     install_requires=[
-        'pyaaf2>=1.4,<1.7',
         'importlib_metadata>=1.4; python_version < "3.8"',
```
[the remark this answers is not in the document]
Comment (JeanChristopheMorinPerso) [reply setup.py:366]: Yes, see https://peps.python.org/pep-0496/#version-numbers.

[the remark this answers is not in the document]
Comment (apetrynet) [reply setup.py:366]: Sorry, I should have been more descriptive. I was thinking about the `< 3.8` part of it.
In light of the comments on the CI fail, this makes sense.

[the remark this answers is not in the document]
Comment (JeanChristopheMorinPerso) [reply setup.py:366]: Ah, I see. We do use `importlib_metadata` on python 3.7, see https://github.com/AcademySoftwareFoundation/OpenTimelineIO/blob/7755f31040a077046485291d514ef272a578826d/src/py-opentimelineio/opentimelineio/plugins/manifest.py#L12-L16.

File: setup.py [hunk 5]
```diff
@@ -342,8 +341,7 @@ def run(self):
```
Comment (JeanChristopheMorinPerso) [inline setup.py:341]: I can't comment on the line, but should we remove `opentimelineio_contrib` from `package_data`?

Comment (apetrynet) [reply setup.py:341]: Hmm, I must have missed that one.

File: docs/tutorials/otio-plugins.md [hunk 6]
```diff
@@ -12,11 +12,10 @@ file should be regenerated.
```
[the remark this answers is not in the document]
Comment (reinecke) [reply docs/tutorials/otio-plugins.md:12]: Let's keep this one for now - I think there's still some useful info and it's all auto-generated.

File: docs/tutorials/quickstart.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ OTIOView has an additional prerequisite to OTIO:
 ## Configure Environment Variables for extra adapters
 
 A default OTIO installation includes only the "Core" adapters, which include CMX EDL, Final Cut Pro 7 XML, and the built in JSON format.
-In order to get access to the "contrib" adapters (which includes the maya sequencer, rv and others), please consult the
+In order to get access to the "contrib" adapters (which includes rv and others), please consult the
 [Adapters documentation page for details](./adapters).
```
[the remark this answers is not in the document]
Comment (reinecke) [reply docs/tutorials/quickstart.md]: I took a pass at this section to return it to relevancy

File: docs/tutorials/adapters.md (hunk 3 above)
Comment (reinecke) [reply docs/tutorials/adapters.md]: I re-wrote this to describe the current adapter landscape and to point readers to the plugin development tutorial.

**Review by apetrynet — APPROVED** [review]
Looks good to me.

Thanks @reinecke !

**Pull request merged**